### PR TITLE
AWS SQS doc

### PR DESCRIPTION
**Pull request by katarzyna-koltun-mx**

### CODE REVIEW

File: content/en/docs/appstore/connectors/aws-sqs.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+#If moving or renaming this doc file, implement a temporary redirect and let the respective team know they should update the URL in the product. See Mapping to Products for more details. 
+---
+
+## 1 Introduction
+
+The [Amazon SQS connector]([#needs-url]) enables your app to connect to other microservices by implementing [Amazon Simple Queue Service (SQS)](https://aws.amazon.com/sqs/).
+
+### 1.1 Typical Use Cases
+
+Amazon SQS helps improve your app by providing a queue service to send messages to other components or microservices in your infrastructure. Implement the Amazon SQS connector if you want your app to have the ability to communicate with other components at scale, for example, to send queue items to a work queue.
```
Comment (katarzyna-koltun-mx) [inline content/en/docs/appstore/connectors/aws-sqs.md]: Q for Riyad & team - is there anything else I can mention about the benefits of using this connector?

File: content/en/docs/appstore/connectors/aws-sqs.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+## 3 Configuration
+
+After you install the connector, you can find it in the **App Explorer**, in the **AmazonSQSConnector** section. The connector provides a [domain model](#domain-model) and several [activities](#activities) that you can use to connect your app to Amazon SQS. Each activity can be implemented by using it in a microflow.
+
+For example, to list all existing Amazon SQS subscriptions, implement the [ListQueue](#list-queues) activity by doing the following steps:
+
+1. In the **App Explorer**, right-click on the name of your module, and then click **Add microflow**.
+2. Enter a name for your microflow, for example, *ACT_ListQueues*, and then click **OK**.
+3. In the new microflow, configure AWS Authentication with static credentials.
+    For more information, see [AWS Authentication](/appstore/connectors/aws-authentication/#static).
```
Comment (katarzyna-koltun-mx) [inline content/en/docs/appstore/connectors/aws-sqs.md]: Q for Riyad & team - Are static credentials recommended here?

File: content/en/docs/appstore/connectors/aws-sqs.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+4. In the **App Explorer**, in the **AmazonSQSConnector** section, find the **ListQueue** activity.
+5. Drag the **ListQueue** activity onto the work area of your microflow.
+6. In the **Toolbox** pane, search for the **Retrieve** activity and drag it onto the microflow area.
+7. Position the **Retrieve** activity between the **ListQueue** activity and the microflow end event.
+8. Double-click the **Retrieve** activity.
+9. In the **Select Association** dialog box, in the **Association** section, click **Select**, and then select **ListQueueResponse** as the association.
```
Comment (katarzyna-koltun-mx) [inline content/en/docs/appstore/connectors/aws-sqs.md]: Q for Riyad & team - Are the activity names correct? Does the microflow make sense? Is there anywhere I could test this?

File: content/en/docs/appstore/connectors/aws-sqs.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+The domain model is a data model that describes the information in your application domain in an abstract way. For more information, see [Domain Model](/refguide/domain-model/).
+
+| Name | Description |
+| --- | --- |
+| `Queue` | Non-persistent; stores the name, region, and URL of a queue |
+| `QueueAttributes` | Non-persistent; stores the attributes for a specific queue, such as the approximate number of messages to be retrieved from the queue |
+| `Message` | Non-persistent; stores the message which is to be sent to a queue, or which is received from an external microservice  |
+| `MessageAttribute` | Non-persistent; stores the key-value pairs for attributes that are to be added to a message |
+
+### 3.2 Activities {#activities}
```
Comment (katarzyna-koltun-mx) [inline content/en/docs/appstore/connectors/aws-sqs.md]: Q for Riyad & team - Are all the parameters correct?

File: content/en/docs/appstore/connectors/aws-sqs.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+The `SendMessage` Amazon SQS activity allows you to send a message to a queue. It requires a message and a queue as parameters.
+
+The input and output for this service are shown in the table below:
+
+| Input | Output |
+| --- | --- |
+| `Message (Object)`, `Queue (Object)` | Send status (Boolean) |
+
+#### 3.2.2 Receive Message
```
Comment (CasperSpronk27) [inline content/en/docs/appstore/connectors/aws-sqs.md]: Receive message can be removed as it wil no longer be included in the connector. This action is duplicated by Receive messages with maxNumberOfMessages set to 1.

Comment (katarzyna-koltun-mx) [reply content/en/docs/appstore/connectors/aws-sqs.md]: @CasperSpronk27 done! Is there anything else? In particular, does the microflow make sense?

File: content/en/docs/appstore/connectors/aws-sqs.md (hunk 2 above)
Comment (CasperSpronk27) [reply content/en/docs/appstore/connectors/aws-sqs.md]: We will be including a GetCredentials action. This action will return either the static or session credentials, based on the constant UseStaticCredentials. Both can be recommended here, which of the two the developer wants to use will depend on their usecase

File: content/en/docs/appstore/connectors/aws-sqs.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+#### 3.2.1 Send Message
+
+The `SendMessage` Amazon SQS activity allows you to send a message to a queue. It requires a message and a queue as parameters.
+
+The input and output for this service are shown in the table below:
+
+| Input | Output |
+| --- | --- |
+| `Message (Object)`, `Queue (Object)` | Send status (Boolean) |
```
Comment (CasperSpronk27) [inline content/en/docs/appstore/connectors/aws-sqs.md]: Input: SendMessageRequest, Credentials
Output: SendMessageResponse

File: content/en/docs/appstore/connectors/aws-sqs.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+| `Message` | Non-persistent; stores the message which is to be sent to a queue, or which is received from an external microservice  |
+| `MessageAttribute` | Non-persistent; stores the key-value pairs for attributes that are to be added to a message |
+
+### 3.2 Activities {#activities}
+
+Activities define the actions that are executed in a microflow or a nanoflow. For the Amazon SQS connector, they represent actions such as sending or receiving a message to or from an SQS queue.
+
+#### 3.2.1 Send Message
+
+The `SendMessage` Amazon SQS activity allows you to send a message to a queue. It requires a message and a queue as parameters.
```
Comment (CasperSpronk27) [inline content/en/docs/appstore/connectors/aws-sqs.md]: Message and queue should be replaced with SendMessageRequest

File: content/en/docs/appstore/connectors/aws-sqs.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+
+The input and output for this service are shown in the table below:
+
+| Input | Output |
+| --- | --- |
+| `Message (Object)`, `Queue (Object)` | Send status (Boolean) |
+
+#### 3.2.2 Receive Messages
+
+The `ReceiveMessages` Amazon SQS activity allows you to receive up to 10 messages from a particular queue. It requires a queue as parameter. By default, this activity uses short polling; for long polling, use the `ReceiveMessageWaitTimeSeconds` parameter.
```
Comment (CasperSpronk27) [inline content/en/docs/appstore/connectors/aws-sqs.md]:
```suggestion
The `ReceiveMessages` Amazon SQS activity allows you to receive up to 10 messages from a particular queue. It requires a ReceiveMessageRequest as parameter. By default, this activity uses short polling; for long polling, use the `WaitTimeSeconds` parameter in the ReceiveMessageRequest entity.
```

File: content/en/docs/appstore/connectors/aws-sqs.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+
+{{% alert color="info" %}}
+If the number of messages in the queue is small (fewer than 1,000), you may receive fewer messages than you requested per a `ReceiveMessage` call. If the number of messages in the queue is very small, you might not receive any messages in a particular `ReceiveMessage` response. If this happens, repeat the request.
+{{% /alert %}}
+
+The input and output for this service are shown in the table below:
+
+| Input | Output |
+| --- | --- |
+| `Queue (Object)` | `Message (List of Objects, maximum 10)` |
```
Comment (CasperSpronk27) [inline content/en/docs/appstore/connectors/aws-sqs.md]:
```suggestion
| `ReceiveMessageRequest` | `ReceiveMessageResponse` |
```

File: content/en/docs/appstore/connectors/aws-sqs.md [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+
+#### 3.2.3 List Queue {#list-queues}
+
+The `ListQueue` Amazon SQS activity allows you to get a list of queues created in a particular region for the user. It does not require any entry parameters.
+
+The input and output for this service are shown in the table below:
+
+| Input | Output |
+| --- | --- |
+| N/A | `Queue (List of Objects, maximum 1000)` |
```
Comment (CasperSpronk27) [inline content/en/docs/appstore/connectors/aws-sqs.md]:
```suggestion
| `ListQueueRequest` | `ListQueueResponse` |
```

File: content/en/docs/appstore/connectors/aws-sqs.md [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+
+The input and output for this service are shown in the table below:
+
+| Input | Output |
+| --- | --- |
+| `Queue (Object)` | `Message (List of Objects, maximum 10)` |
+
+#### 3.2.3 List Queue {#list-queues}
+
+The `ListQueue` Amazon SQS activity allows you to get a list of queues created in a particular region for the user. It does not require any entry parameters.
```
Comment (CasperSpronk27) [inline content/en/docs/appstore/connectors/aws-sqs.md]:
```suggestion
The `ListQueue` Amazon SQS activity allows you to get a list of queues created in a particular region for the user. It requires a ListQueueRequest parameter as an input.
```

File: content/en/docs/appstore/connectors/aws-sqs.md [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+
+The input and output for this service are shown in the table below:
+
+| Input | Output |
+| --- | --- |
+| N/A | `Queue (List of Objects, maximum 1000)` |
+
+#### 3.2.4 Delete Message
+
+The `DeleteMessage` Amazon SQS activity allows you to delete a message from a queue. It requires a queue and a message to be deleted as parameters.
```
Comment (CasperSpronk27) [inline content/en/docs/appstore/connectors/aws-sqs.md]:
```suggestion
The `DeleteMessage` Amazon SQS activity allows you to delete a message from a queue. It requires a DeleteMessageRequest as parameter.
```

File: content/en/docs/appstore/connectors/aws-sqs.md [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+
+The input and output for this service are shown in the table below:
+
+| Input | Output |
+| --- | --- |
+| `Message (Object)`, `Queue (Object)` | Delete status (Boolean) |
```
Comment (CasperSpronk27) [inline content/en/docs/appstore/connectors/aws-sqs.md]:
```suggestion
| `DeleteMessageRequest` | Delete status (Boolean) |
```

File: content/en/docs/appstore/connectors/aws-sqs.md [hunk 14]
```diff
@@ -0,0 +1,148 @@
...
+
+#### 3.2.5 CreateQueue
+
+The `CreateQueue` Amazon SQS activity allows you to create a queue in a specific region with the credentials provided. It requires that you authenticate as an AWS user with admin credentials.
+
+The input and output for this service are shown in the table below:
+
+| Input | Output |
+| --- | --- |
+| `Queue (Object)` | Create status (Boolean) |
```
Comment (CasperSpronk27) [inline content/en/docs/appstore/connectors/aws-sqs.md]:
```suggestion
| `CreateQueueRequest` | Create status (Boolean) |
```

File: content/en/docs/appstore/connectors/aws-sqs.md [hunk 15]
```diff
@@ -0,0 +1,148 @@
...
+
+#### 3.2.6 Get Queue Attributes
+
+The `GetQueueAttributes` Amazon SQS activity allows you to view all attribute values set for a specific queue. It requires that you authenticate as an AWS user with admin credentials.
+
+The input and output for this service are shown in the table below:
+
+| Input | Output |
+| --- | --- |
+| `Queue (Object)` | `QueueAttributes (Object)` |
```
Comment (CasperSpronk27) [inline content/en/docs/appstore/connectors/aws-sqs.md]:
```suggestion
| `GetQueueAttributesRequest` | `GetQueueAttributesResponse`| 
```

File: content/en/docs/appstore/connectors/aws-sqs.md [hunk 16]
```diff
@@ -0,0 +1,148 @@
...
+
+#### 3.2.7 Purge Queue
+
+The `PurgeQueue` Amazon SQS activity allows you to delete all messages from a specific queue. It requires that you authenticate as an AWS user with admin credentials.
+
+The input and output for this service are shown in the table below:
+
+| Input | Output |
+| --- | --- |
+| `Queue (Object)` | `Purge status (Boolean)` |
```
Comment (CasperSpronk27) [inline content/en/docs/appstore/connectors/aws-sqs.md]:
```suggestion
| `PurgeQueueRequest` | `Purge status (Boolean)` |
```

File: content/en/docs/appstore/connectors/aws-sqs.md [hunk 17]
```diff
@@ -0,0 +1,148 @@
...
+
+#### 3.2.8 Delete Queue
+
+The `DeleteQueue` Amazon SQS activity allows you to delete a specific queue. It requires that you authenticate as an AWS user with admin credentials.
+
+The input and output for this service are shown in the table below:
+
+| Input | Output |
+| --- | --- |
+| `Queue (Object)` | `Delete status (Boolean)` |
```
Comment (CasperSpronk27) [inline content/en/docs/appstore/connectors/aws-sqs.md]:
```suggestion
| `DeleteQueueRequest` | `Delete status (Boolean)` |
```

File: content/en/docs/appstore/connectors/aws-sqs.md (hunk 4 above)
Comment (CasperSpronk27) [reply content/en/docs/appstore/connectors/aws-sqs.md]: Message sent to Kasia to update the domain model

File: content/en/docs/appstore/connectors/aws-sqs.md (hunk 2 above)
Comment (katarzyna-koltun-mx) [reply content/en/docs/appstore/connectors/aws-sqs.md]: Done

File: content/en/docs/appstore/connectors/aws-sqs.md (hunk 7 above)
Comment (katarzyna-koltun-mx) [reply content/en/docs/appstore/connectors/aws-sqs.md]: Done.

File: content/en/docs/appstore/connectors/aws-sqs.md (hunk 6 above)
Comment (katarzyna-koltun-mx) [reply content/en/docs/appstore/connectors/aws-sqs.md]: Done.

**Comment by katarzyna-koltun-mx** [comment]
Originally an old PR that I'm reworking to fit the new template.

**Comment by katarzyna-koltun-mx** [comment]
Ready for the team.

**Comment by katarzyna-koltun-mx** [comment]
@nicokvmendix made a new PR forked from this one: https://github.com/mendix/docs/pull/6386 - I'll close mine.

**Pull request closed without merging**